deno/fhirchat-demo/islands/ChatIsland.tsx

```diff
@@ -1,43 +1,54 @@
-import { useContext, useEffect } from "preact/hooks";
-import { AIConversationState } from "../hooks/aiConversationContext.ts";
+import { useContext, useEffect, useMemo } from "preact/hooks";
+import { signal } from "@preact/signals";
+import { AIConversationContext } from "../hooks/aiConversationContext.ts";
+import { Interaction, Message } from "../types/conversation.ts";
+import { extractInteractions } from "../helpers/extractConversationInteractions.ts";
+import UserMessage from "../components/UserMessage.tsx";
+import AgentMessage from "../components/AgentMessage.tsx";
 import { useTextWithTypeAnimation } from "../hooks/typingAnimation.ts";
 
 export default function ChatIsland() {
   const {
-    question,
     conversation,
-    setQuestion,
-    appendToConversation,
-    submitQuestion,
     closeConversation,
-  } = useContext(AIConversationState);
+    submitQuestion,
+  } = useContext(AIConversationContext);
 
-  const userQuestion = conversation.value[0];
-  const agentResponse = conversation.value[1];
+  const interactions: Array<Interaction> = useMemo(() =>
+    extractInteractions(
+      conversation.value,
+    ), [conversation.value]);
+
+  const userInput = signal<string>("");
```

Comment:
Renamed to `userInput` because it's a bit less ambiguous than `question`

Reply:
definitely, good renaming :)